vulpes/hooks/save_load.cpp

```diff
@@ -0,0 +1,119 @@
+/*
+ * Vulpes (c) 2020 gbMichelle
+ *
+ * This program is free software under the GNU General Public License v3.0 or later. See LICENSE for more information.
+ */
+
+#include <windows.h>
+
+#include <hooker/hooker.hpp>
+#include <hooker/function_pointer_safe.hpp>
+
+#include <vulpes/memory/signatures.hpp>
+
+#include "save_load.hpp"
+
+DEFINE_EVENT_HOOK_LIST(EVENT_BEFORE_SAVE, before_save_events);
+DEFINE_EVENT_HOOK_LIST(EVENT_BEFORE_LOAD, before_load_events);
+DEFINE_EVENT_HOOK_LIST(EVENT_AFTER_LOAD,  after_load_events);
+
+static bool doing_core = false;
+
+typedef void (*fp_type)();
+static void (*before_save_proc_orig)();
+static void (*before_load_proc_orig)();
+static void (*after_load_proc_orig)();
+static int (*core_load_orig)(char*);
+
+static void before_save_proc_hook() {
+    call_in_order(before_save_events);
+    exec_if_valid(before_save_proc_orig);
+}
+
+static void before_load_proc_hook() {
+    if (!doing_core) {
+        call_in_order(before_load_events);
+    }
+    exec_if_valid(before_load_proc_orig);
+    // When this function finishes executing the vanilla gamestate is loaded
+    // from disk/in-memory copy.
+}
+
+static void after_load_proc_hook() {
+    exec_if_valid(after_load_proc_orig);
+    if (!doing_core) {
+        call_in_order(after_load_events);
+    }
+}
+
+// https://github.com/Sigmmma/Vulpes/issues/68
+// This function was a wrapper to avoid calling checkpoint load logic during
+// a core_load. Because there is distinction.
+// But core loading and saving is kind of a hack.
+__attribute__((regparm(1)))
+static int core_load_hook(char* name) {
+    cprintf_error("core_load is disabled");
+    //doing_core = true;
+    //int result = core_load_orig(name);
+    //doing_core = false;
+    //cprintf_info("After core_load ping!");
+    return 0;//result;
+}
+
+static Patch(
+    core_load_hook_patch, NULL, 5,
+    CALL_PATCH, &core_load_hook
+);
+
+static bool hook_save_load_initialized = false;
+
+void init_save_load_hook() {
+    if (!hook_save_load_initialized) {
```

Comment:
Minor stylistic choice, but you can invert the logic here and exit early with a return, then unindent the rest of the function.

Reply:
I can't return early because then I skip executing a core game function with exec_if_valid. Adding a comment.